Kernel/SpecialOrbits.m

```diff
@@ -285,6 +287,24 @@
 	If[KerrGeoBoundOrbitQ[0,p,e,1] == KerrGeoScatterOrbitQ[0,p,e,1] == False, True, False]
 
 
+(* ::Section::Closed:: *)
+(*OnSeparatrixQ*)
+
+
+(* ::Text:: *)
+(*Test to see if the orbit is exactly on the separatrix, where many of our expressions become singular. This is a stopgap solution until we can think of a nice way of taking the separatrix limit of all of these functions. *)
+
+
+KerrGeoOnSeparatrixQ[a_?NumericQ,p_?NumericQ,e_?NumericQ,x_?NumericQ]:= Module[{En,L,Q,r1,r2,r3,r4,prec},
+
+prec = Precision[{a,p,e,x}];
+{En,L,Q} = Values[KerrGeoConstantsOfMotion[a,p,e,x]];
+{r1,r2,r3,r4} = KerrGeodesics`OrbitalFrequencies`Private`KerrGeoRadialRoots[a, p, e, x, En, Q];
+(*Assuming KerrGeoSeparatrix only finds solution to within half of the working precision*)
```

Comment:
We only encounter singularities if the roots are exactly equal, so perhaps we don't need a tolerance here?

Reply:
There needs to be a tolerance. If you get rid of it, it will tell you that you are not on the separatrix, but when you go to evaluate say, the frequencies, they will all blow up. Try the following test:

```mathematica
atest = 0.9;
etest = 0.3;
xtest = 0.8;

ptest = KerrGeoSeparatrix[atest, etest, xtest];

{r1, r2, r3, r4} = 
 KerrGeodesics`OrbitalFrequencies`Private`KerrGeoRadialRoots[atest, 
  ptest, etest, xtest];

r2-r3
```
Output will be close to zero, but not exactly. 
If one then evaluates the frequencies at this point, they blow up. Same is true of higher precision numbers too. 
This comes down to the fact you have to use numerical root-finding to get the separatrix location for anything more complicated than eccentric, equatorial Schwarzschild.

Reply:
Ok, it sounds like a tolerance is appropriate here, then.